app/src/main/java/org/rudirect/android/util/RUDirectUtil.java

```diff
@@ -44,4 +46,20 @@ public static int dpToPx(int dp) {
         DisplayMetrics displayMetrics = RUDirectApplication.getContext().getResources().getDisplayMetrics();
         return (int) ((dp * displayMetrics.density) + 0.5);
     }
+
+    // Gets stop nearest to location.
+    public static BusStop getNearestStop(Location location) {
+        double minDist = Double.MAX_VALUE;
+        BusStop closestStop = null;
+        for (BusStop stop : RUDirectApplication.getBusData().getBusStops()) {
+            double dist = Math.sqrt(Math.pow(Double.parseDouble(stop.getLatitude()) - location.getLatitude(), 2) +
```

Comment:
Math.pow is a slow operation, so avoid using it for powers of 2. Instead, store the first argument in a variable and just multiply it by itself.